Entities/Deployment.h

```diff
@@ -325,7 +325,7 @@ ClassInfoGetters
 // Arguments:       None.
 // Return value:    None.
 
-	virtual void NewID() { m_ID = SelectRand(1, 0xFFFF); };
+	virtual void NewID() { m_ID = RandomNum<int>(1, 0xFFFF); };
```

Comment:
I have no idea about black template magic, but the template should automagically choose the right uhhh... overload I guess... depending on the passed in parameter types.

Calling `RandomNum(0,0)` will use `typename std::enable_if<std::is_integral<intType>::value, intType>::type RandomNum(intType min, intType max)` so you don't really need to specify `<int>`.

While calling `RandomNum(0.0,0.0)` will use `typename std::enable_if<std::is_floating_point<floatType>::value, floatType>::type RandomNum(floatType min, floatType max)`

Passing in mismatching parameter types will squiggle as it should, and in this case specifying `<int>` solves that.

So yeah, no need to specify `<int>` where there aren't any parameter type mismatches (e.g. `RandomNum(1, 0xFFFF)` is ok because both are `int`, but `RandomNum(1, moduleList.size() - 1)` is not because it's `int` and `size_type`).


Applies everywhere.

Reply:
Ok, removed the <int> where it isn't required.